src/ISovryn.ts

```diff
@@ -498,3 +503,36 @@ export function handleRollover(event: RolloverEvent): void {
   rolloverEntity.transaction = event.transaction.hash.toHexString()
   rolloverEntity.save()
 }
+
+export function handleWithdrawFees(event: WithdrawFeesEvent): void {
+  createAndReturnTransaction(event)
+  function createWithdrawFees(amount: BigDecimal, token: Token, feeType: string, event: ethereum.Event): void {
+    const withdrawFees = new ProtocolWithdrawFee(event.transaction.hash.toHexString() + '-' + event.logIndex.toHexString() + '-' + feeType)
+    withdrawFees.amount = amount
+    withdrawFees.amountUsd = amount.times(token.lastPriceUsd)
+    withdrawFees.token = token.id
+    withdrawFees.feeType = feeType
+    withdrawFees.transaction = event.transaction.hash.toHexString()
+    withdrawFees.timestamp = event.block.timestamp.toI32()
+    withdrawFees.emittedBy = event.address.toHexString()
+    withdrawFees.save()
+  }
+  const token = Token.load(event.params.token.toHexString())
+  if (token != null) {
+    createWithdrawFees(decimal.fromBigInt(event.params.tradingAmount, DEFAULT_DECIMALS), token, ProtocolFeeType.Trading, event)
```

Comment:
not decimals from token entity instead of default?

Reply:
I'll change to token decimals since token entity is already loaded